### DISCUSSION TITLE
How to restore previous tasks?

### DISCUSSION THREAD

**Discussion by Ocean-OS** [minbrowser/min#2527] [Q&A]
I was working in an in-browser IDE and the browser crashed. I tried to go back to the page in my history, but the code was saved in the URL hash and the URL that was restored was the original version of the code from when I first opened the IDE. 
This also removed the previous task, so I cannot restore my progress (which was about 2-3 hours and 4-500 lines of code). 
Is there a way to restore the previous (not visible) tasks?

**Comment by PalmerAL** [comment]
Hi, sorry to hear that. Unfortunately, I'm guessing there probably isn't. Some places you could look would be pressing shift+cmd+t (which reopens previously closed tabs), or the history list (click the clock icon in the searchbar), but it sounds like most likely the updated hash was never added to history.

What IDE is this? Storing important data in the URL hash generally sounds like a really bad idea, so I'd recommend trying to find a different IDE if you can, but it also sounds like perhaps we have some bug where we're not updating the history entry when the hash changes.

**Reply by Ocean-OS** [reply]
The [IDE](https://svelte.dev/repl) is more like a small playground, not really meant for actual projects (and it's my fault for using it instead of something meant for projects), but it's definitely a bug in the browser that it didn't save the updated URL.
